SIT.Manager.Avalonia/ViewModels/MainViewModel.cs

```diff
@@ -6,41 +6,87 @@
 using CommunityToolkit.Mvvm.Messaging;
 using FluentAvalonia.UI.Controls;
 using FluentAvalonia.UI.Media.Animation;
-using SIT.Manager.Avalonia.Interfaces;
 using SIT.Manager.Avalonia.Models;
 using SIT.Manager.Avalonia.Models.Messages;
 using System;
 using System.Collections.ObjectModel;
 using System.Diagnostics;
 using System.IO;
 using System.Threading.Tasks;
+using System.Net.Http;
+using System.Reflection;
+using Microsoft.Extensions.Logging;
+using ReactiveUI;
+using System.Reactive.Disposables;
+using SIT.Manager.Avalonia.ManagedProcess;
 
 namespace SIT.Manager.Avalonia.ViewModels;
 
 public partial class MainViewModel : ViewModelBase, IRecipient<PageNavigationMessage>
 {
+    private const string MANAGER_VERSION_URL = @"https://raw.githubusercontent.com/stayintarkov/SIT.Manager.Avalonia/master/VERSION";
     private readonly IActionNotificationService _actionNotificationService;
     private readonly IBarNotificationService _barNotificationService;
+    private readonly IManagerConfigService _managerConfigService;
+    private readonly ILogger<MainViewModel> _logger;
+    private readonly HttpClient _httpClient;
 
     private Frame? contentFrame;
 
     [ObservableProperty]
     private ActionNotification? _actionPanelNotification = new(string.Empty, 0, false);
 
+    [ObservableProperty]
+    private bool _updateAvailable = false;
+
     public ObservableCollection<BarNotification> BarNotifications { get; } = [];
 
     public IAsyncRelayCommand UpdateButtonCommand { get; }
+    public IAsyncRelayCommand CloseButtonCommand { get; }
 
-    public MainViewModel(IActionNotificationService actionNotificationService, IBarNotificationService barNotificationService) {
+    public MainViewModel(IActionNotificationService actionNotificationService,
+        IBarNotificationService barNotificationService,
+        IManagerConfigService managerConfigService,
+        ILogger<MainViewModel> logger,
+        HttpClient httpClient)
+    {
         _actionNotificationService = actionNotificationService;
         _barNotificationService = barNotificationService;
+        _managerConfigService = managerConfigService;
+        _logger = logger;
+        _httpClient = httpClient;
 
         _actionNotificationService.ActionNotificationReceived += ActionNotificationService_ActionNotificationReceived;
         _barNotificationService.BarNotificationReceived += BarNotificationService_BarNotificationReceived;
 
         WeakReferenceMessenger.Default.Register(this);
 
         UpdateButtonCommand = new AsyncRelayCommand(UpdateButton);
+        CloseButtonCommand = new AsyncRelayCommand(() => { UpdateAvailable = false; return Task.CompletedTask; });
```

Comment:
This could just be a normal RelayCommand as we don't use any asynchronous methods in it.